hudi-common/src/main/java/org/apache/hudi/common/util/CleanerUtils.java

```diff
@@ -85,8 +85,9 @@ public static HoodieCleanMetadata convertCleanMetadata(String startCleanTime,
         partitionBootstrapMetadataMap.put(stat.getPartitionPath(), bootstrapMetadata);
       }
       totalDeleted += stat.getSuccessDeleteFiles().size();
-      if (earliestCommitToRetain == null) {
-        // This will be the same for all partitions
+
+      // Use the largest completion time.
+      if (needUpdate(stat.getEarliestCommitToRetain(), earliestCommitToRetain)) {
         earliestCommitToRetain = stat.getEarliestCommitToRetain();
         lastCompletedCommitTimestamp = stat.getLastCompletedCommitTimestamp();
```

Comment:
Should we use `stat.getLastCompletedCommitTimestamp()` for both variables?

Reply:
yes i think we should use completion time for both